src/modules/esl-popup/core/esl-popup.ts

```diff
@@ -186,15 +172,15 @@ export class ESLPopup extends ESLToggleable {
     return $arrow;
   }
 
-  /**
-   * Actions to execute before showing of popup. Handles the activator and updates the position of the popup.
-   * @returns false if the show task should be canceled
-   */
+  /** Runs additional actions on show popup request */
   protected override onBeforeShow(params: ESLToggleableActionParams): boolean | void {
-    if (this.open) this.afterOnHide();
     this.activator = params.activator;
-    if (this.open) this.afterOnShow();
-    if (this.open && !params.force) return false; // the show task will be forced to run so the next steps are unnecessary
+    if (this.open) {
+      this.afterOnHide();
```

Comment:
Is it safe to unsubscribe in the `afterOnHide()` method when the activator has already changed?

Reply:
There is no longer required, ESL Event listeners knows exactly what was their targets on subscription